File: Models/Employer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MtpApp.Models
{
    public class Employer
    {
        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        [Display(Name = "Company Name")]
        public string Name { get; set; }

        [Required]
        [Display(Name = "Sector")]
        public int JobCategoryId { get; set; }

        public JobCategory JobCategory { get; set; }

        [StringLength(510)]
        public string Address { get; set; }

        [Required]
        [StringLength(255)]
        [Display(Name = "Contact Person ")]
        public string ContactPerson { get; set; }

        [StringLength(255)]
        [Display(Name = "Contact Phone")]
        public string ContactPhone { get; set; }    

        [StringLength(255)]
        public string Email { get; set; }

        [StringLength(255)]
        public string Website { get; set; }

        [StringLength(510)]
        public string Note { get; set; }

        [StringLength(255)]
        public string Branch { get; set; }

        public DateTime CorporateDate { get; set; }

        public string Status { get; set; }
    }
}

File: spring-backend/src/main/java/com/mtp/api/models/Employer.java
package com.mtp.api.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "Employers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Employer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    @Size(max = 255)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "JobCategoryId")
    private JobCategory jobCategory;

    @Size(max = 510)
    private String address;

    @NotNull
    @Size(max = 255)
    private String contactPerson;

    @Size(max = 255)
    private String contactPhone;

    @Size(max = 255)
    private String email;

    @Size(max = 255)
    private String website;

    @Size(max = 510)
    private String note;

    @Size(max = 255)
    private String branch;

    private LocalDateTime corporateDate;

    private String status;

    private String logoUrl;
}
